test/realtime_web/plugs/rls_authorization_test.exs

```diff
@@ -0,0 +1,96 @@
+defmodule RealtimeWeb.RlsAuthorizationTest do
+  use RealtimeWeb.ConnCase
+
+  import Plug.Conn
+
+  alias Realtime.Tenants
+  alias RealtimeWeb.Joken.CurrentTime
+  alias RealtimeWeb.RlsAuthorization
+
+  setup context do
+    start_supervised!(CurrentTime.Mock)
+    tenant = tenant_fixture()
+
+    settings = Realtime.PostgresCdc.filter_settings("postgres_cdc_rls", tenant.extensions)
+    settings = Map.put(settings, "id", tenant.external_id)
+    settings = Map.put(settings, "db_socket_opts", [:inet])
+
+    start_supervised!({Tenants.Migrations, settings})
+
+    {:ok, db_conn} = Tenants.Connect.lookup_or_start_connection(tenant.external_id)
+
+    clean_table(db_conn, "realtime", "channels")
+    channel_fixture(tenant, context.rls_setup_params || %{})
+    create_rls_policy(db_conn, context.rls, context.rls_setup_params)
+
+    claims = %{sub: random_string(), role: "authenticated", exp: Joken.current_time() + 1_000}
+    signer = Joken.Signer.create("HS256", "secret")
+
+    jwt = Joken.generate_and_sign!(%{}, claims, signer)
+
+    %{jwt: jwt, claims: claims, tenant: tenant}
+  end
+
+  @select_authenticated_role_tests [
+    {"anon", :select_authenticated_role, nil, [], %{read: false}},
+    {"authenticated", :select_authenticated_role, nil, [], %{read: true}}
+  ]
+  @select_authenticated_role_on_channel_name_tests [
+    {
+      "authenticated",
+      :select_authenticated_role_on_channel_name,
+      %{name: random_string()},
+      [{"id", "1"}],
+      %{read: true}
+    },
+    {
+      "anon",
+      :select_authenticated_role_on_channel_name,
+      %{name: random_string()},
+      [{"id", "1"}],
+      %{read: false}
+    }
+  ]
+
+  for {role, rls, rls_setup_params, conn_params, %{read: expected_read}} <-
+        @select_authenticated_role_tests ++ @select_authenticated_role_on_channel_name_tests do
+    @tag role: role, rls: rls, rls_setup_params: rls_setup_params
+    test "assigns the permissions correctly to #{role} and #{rls} rule", %{
```

Comment:
Consider an alternative where you create a small helper in your test and share it. Something like:

```
test "assigns the permission correctly to anon and authenticated role", context do
  assert_foo_bar_baz(context, %{name: random_string()}, params, ...)
end
```

You should have a bit of repetition but you should be able to get rid of most meta-programming, which may or may not make them more maintainable. I thought I would mention it as an option.